src/client.ts

```diff
@@ -0,0 +1,494 @@
+    this.notification_resolvables.delete(method);
+  }
+
+  /**
+   * Gets the text for the given selector
+   * Must be an input element
+   *
+   * @param selector - eg input[type="submit"] or #submit
+   *
+   * @throws When:
+   *     - Error with the element (using selector)
+   *
+   * @returns The text inside the selector, eg could be "" or "Edward"
+   */
+  public async getInputValue(selector: string): Promise<string> {
+    const command = `document.querySelector('${selector}').value`;
+    const res = await this.sendWebSocketMessage("Runtime.evaluate", {
+      expression: command,
+    }) as {
+      result: {
+        type: "undefined" | "string";
+        value?: string;
+      };
+    } | { // Present if we get a `cannot read property 'value' of null`, eg if `selector` is `input[name="fff']`
+      result: Exception;
+      exceptionDetails: ExceptionDetails;
+    };
+    if ("exceptionDetails" in res) {
+      this.checkForErrorResult(res, command);
+    }
+    const type = (res as DOMOutput).result.type;
+    if (type === "undefined") { // not an input elem
+      return "undefined";
+    }
+    // Tried and tested, value and type are a string and `res.result.value` definitely exists at this stage
+    const value = (res.result as { value: string }).value;
+    return value || "";
+  }
+
+  /**
+   * Close/stop the sub process, and close the ws connection. Must be called when finished with all your testing
+   */
+  public async done(): Promise<void> {
+    // [Dirty fix 1] Dirty hack... There is a bug with WS API that causes async ops. I (ed) have found that closing the conn inside the message handler fixes it, which is why we are trigger a message here, so the `onmessage` handler can close the connection
+    if (this.socket.readyState !== 3) { // Conditional here, as this method cna be called by the assertion methods, so if an assertion method has failed, and the user calls `.done()`, we wont try close an already close websocket
+      const p = deferred();
+      this.socket.onclose = function () {
+        p.resolve();
+      };
+      this.socket.send(JSON.stringify({
+        id: -1,
+        method: "DOM.getDocument", // Can be anything really, we just wanna trigger an event
+      }));
+      // Then wait for the promise to be resolved when the WS client is done
+      await p;
+    }
+    if (this.browser_process_closed === false) {
+      this.browser_process.stderr!.close();
+      this.browser_process.stdout!.close();
+      this.browser_process.close();
+      this.browser_process_closed = true;
+    }
+    // Because if using firefox and windows, the close on the browser subprocess doesn't actually close the processes
+    if (Object.getPrototypeOf(this.constructor).name === "FirefoxClient" && Deno.build.os === "windows") {
+      const p = Deno.run({
+        cmd: ["taskkill", "/F", "/IM", "firefox.exe"],
+        stdout: "null",
+        stderr: "null"
+      })
+      await p.status()
+      p.close()
```

Comment:
try:

- removing this
- `await browser.status()` above
- see if that fixes the issue
- if not, keep in the above logic and open an issue with deno